### ISSUE TITLE
Run Linux Web Engine builds on machines with more cores

### ISSUE THREAD

**Issue by zanderso** [flutter/flutter#114722]
`Linux Web Engine` builds are running in ~50 minutes, which is substantially over the goal of 30 minutes. One reason for this is that the `wasm_release` build is not goma-enabled. Even so, the `ninja` command to build the `wasm_release` build is still passing a `-j` of 200, which is way beyond the 8 cores that the Linux VMs appear to have. We need to do two things:
1. Use `autoninja` rather than `ninja -j200` to build `wasm_release`
2. Run the `Linux Web Engine` build on a machine with as many cores as possible.

cc @keyonghan @eyebrowsoffire

@eyebrowsoffire Do we (or do we plan to in the next O(weeks)) do a `wasm_release` build on any configurations aside from `Linux Web Engine`?

**Comment by zanderso** [comment]
I'll take a look at switching to `autoninja`.

**Comment by eyebrowsoffire** [comment]
Previously `autoninja` was giving us issues locally for people with goma enabled, since it massively parallelized the ninja build to way too many jobs, assuming goma would pick up the jobs. However, goma doesn't work with emscripten. We fixed this in `//flutter/tools/gn` by just disabling goma for the wasm build, since it can't take advantage of goma: https://github.com/flutter/engine/pull/37294

However, from your description here it sounds sort of like the opposite setup? Goma is disabled, but explicitly running with `-j200`? That seems very wrong. If we switch to `autoninja` or just let `ninja` pick a sane default, the build should take nowhere near 50 minutes, it should be much quicker.

**Comment by eyebrowsoffire** [comment]
I also was planning on making a bunch of changes to the CI recipes anyway in order to *only* build wasm_release and not do the host build at all for the web engine tests: https://github.com/flutter/flutter/issues/113303

Maybe it makes sense to just cut over and immediately do that anyway?

**Comment by zanderso** [comment]
Switching to autoninja is pretty simple: https://flutter-review.googlesource.com/c/recipes/+/35660

**Comment by godofredoc** [comment]
Change to not clean source code caches in web engine:

https://flutter-review.googlesource.com/c/recipes/+/35661

**Comment by keyonghan** [comment]
Yeah. We just have a TODO action to monitor the queue time of these 32 core bots to make sure we are within SLO.
